src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -1569,7 +1581,7 @@ public async Task SetWorkingDirectoryAsync(string path)
         /// <param name="isPathAlreadyEscaped">Specify false to have the path escaped, otherwise specify true if the path has already been escaped.</param>
         public async Task SetWorkingDirectoryAsync(string path, bool isPathAlreadyEscaped)
         {
-            this.InitialWorkingDirectory = path;
+            this.InitialWorkingDirectory = path ?? throw new ArgumentNullException(nameof(path));
```

Comment:
Use the `Validate.IsNull`?

Reply:
Missed this initially, but how do we feel about these methods?

They kind of mess with the call stack a little bit.  It's not a huge deal, but it does make debugging *slightly* more tedious since you need to pick the right frame.  And it makes `Exception.TargetSite` kinda useless.

Obviously I'm not expecting anything in this merged PR to change, just bringing it up in case any of y'all are also leaning that way. WTB macros 🙁

Reply:
The Util method or the SetWorkingDirectory?

Reply:
The Util methods for argument validation

Reply:
Yeah they do seem very unnecessary and even have drawbacks as you've pointed out.

I'd be fine with them being removed but also not a huge problem atm